Source/WebKit/UIProcess/WebAuthentication/Cocoa/WebAuthenticatorCoordinatorProxy.mm

```diff
@@ -369,9 +369,12 @@ static inline ASAuthorizationPublicKeyCredentialLargeBlobSupportRequirement toAS
             WebCore::ExceptionData exceptionData = { ExceptionCode::NotAllowedError, @"" };
             WebCore::AuthenticatorAttachment attachment = AuthenticatorAttachment::Platform;
             if ([error.get().domain isEqualToString:ASAuthorizationErrorDomain] && error.get().code == ASAuthorizationErrorFailed && error.get().userInfo[NSUnderlyingErrorKey]) {
-                RetainPtr<NSError> underlyingError = adoptNS(error.get().userInfo[NSUnderlyingErrorKey]);
-                if ([underlyingError.get().domain isEqualToString:ASCAuthorizationErrorDomain] && underlyingError.get().code == ASCAuthorizationErrorSecurityError)
-                    exceptionData = { ExceptionCode::SecurityError, underlyingError.get().userInfo[NSLocalizedFailureReasonErrorKey] };
+                RetainPtr<NSError> underlyingError = retainPtr(error.get().userInfo[NSUnderlyingErrorKey]);
+                if ([underlyingError.get().domain isEqualToString:ASCAuthorizationErrorDomain] && underlyingError.get().code == ASCAuthorizationErrorSecurityError) {
+                    RetainPtr underlyingUserInfo = retainPtr(underlyingError.get().userInfo);
+                    RetainPtr failureReason = retainPtr(underlyingUserInfo.get()[NSLocalizedFailureReasonErrorKey]);
+                    exceptionData = { ExceptionCode::SecurityError, failureReason.get() };
```

Comment:
I didn't check the type of `exceptionData` but would `WTFMove(failureReason)` reduce churn here?

Reply:
The second member of `exceptionData` is a WTFString. I don't see a way in `WTFString.h` for a RetainPtr<NSString>&&  to work here.